packages/core/src/lib/markdown-it/index.js

```diff
@@ -75,31 +75,77 @@ markdownIt.renderer.rules.fence = (tokens, idx, options, env, slf) => {
   if (highlightLinesInput) {
     const highlightLines = highlightLinesInput.split(HIGHLIGHT_LINES_DELIMITER_REGEX);
     highlightRules = highlightLines
-      .map(ruleStr => HighlightRule.parseRule(ruleStr, -startFromZeroBased, lines))
+      .map(ruleStr => HighlightRule.parseRule(ruleStr, -startFromZeroBased, str.split('\n')))
       .filter(rule => rule); // discards invalid rules
   }
 
   if (lang && hljs.getLanguage(lang)) {
     try {
-      /* We cannot syntax highlight THEN split by lines. For eg:
+      /* With highlightjs version >= v10.7.0, usage of continuation is deprecated
+
+      For the purposes of line-by-line highlighting, we have to first highlight the
+      whole block, then split the resulting html string according to '\n', and add
+      the corresponding opening and closing tags for the html string to be well-formed and
+      maintain the correct state per line.
+      eg:
       ```markdown
       *****
       -----
       ```
 
       becomes
 
-      <span class="hljs-section">*****
-      -----</span>
-      Note the line break contained inside a <span> element.
-      So we have to split by lines THEN syntax highlight.
+      "<span class="hljs-section"><span class="hljs-strong">****</span>
+      <span class="hljs-emphasis">*\n-----\n</span></span>"
+
+      after splitting on '\n':
+
+      ["<span class="hljs-strong">****</span><span class="hljs-emphasis">*",
+       "-----",
+       "</span>",
+      ]
+
+      tokenStack maintains the visited tokens. For eg, for <span class="xyz">, the token would be "xyz".
+      If it encounters a </span>, it pops from the stack.
+      Else, it pushes the token to the stack.
+
+      At the end of each line, it will be prepended by the necessary
+      opening tags and appended by the neccessary closing tags.
+
+      After processing, the lines array should be
+
+      ["<span class="hljs-strong">****</span><span class="hljs-emphasis">*</span>",
+       "<span class="hljs-emphasis">-----</span>",
+       "<span class="hljs-emphasis"></span>",
+      ]
        */
-      // state stores the current parse state of hljs, so that we can pass it on line by line
-      let state = null;
+      lines = hljs.highlight(str, { language: lang, ignoreIllegals: true }).value;
+      const tokenStack = [];
+
+      // remove <span class="hljs-section"> ... </span>
+      // as the whole section would be wrapped by html tags at the end
+      if (lines.includes('hljs-section')) {
+        lines = lines.slice(27);
+        lines = lines.slice(0, lines.length - 7);
```

Comment:
Magic numbers here. For easier understanding, better to put the lengths of `<span class="hljs-section">` and its closing counterpart somewhere, not as a direct number, but derived from the length of the respective strings.